File: mywebsite-parent/mywebsite-manager-web/src/main/webapp/angularjs/js/service/trendsService.js
app.service('trendsService', function ($http) {

    /**
     * 条件分页搜索
     * @param searchEntity
     * @param page
     * @param rows
     * @returns {*}
     */
    this.search = function (searchEntity, page, rows) {
        return $http.post('../trends/search.do?page=' + page + '&rows=' + rows, searchEntity);
    };

    /**
     * 根据id查询动态
     * @param id
     * @returns {*}
     */
    this.findOneById = function (id) {
        return $http.get('../trends/findOneById.do?id=' + id);
    };

    /**
     * 更新动态
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../trends/update.do', entity);
    };

    /**
     * 添加动态
     * @param entity
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../trends/add.do', entity);
    };

    /**
     * 批量删除
     * @param ids
     * @returns {*}
     */
    this.deleteIds = function (ids) {
        return $http.get('../trends/deleteIds.do?ids='+ids);
    };

    /**
     * 批量发表
     * @param ids
     * @param status
     * @returns {*}
     */
    this.updateStatusIds = function (ids,status) {
        return $http.get('../trends/updateStatusIds.do?ids='+ids+'&status='+status);
    };
});

File: mywebsite-parent/mywebsite-content-interface/src/main/java/cn/boom/mywebsite/content/service/TrendsService.java
package cn.boom.mywebsite.content.service;

import cn.boom.mywebsite.pojo.MyWebSiteResult;
import cn.boom.mywebsite.pojo.TbTrends;

public interface TrendsService {

    /**
     * 根据id查询动态记录
     * @param id
     * @return
     */
    public abstract TbTrends findOneById(Long id);

    /**
     * 条件分页搜索
     * @param tbTrends
     * @param pageNum
     * @param pageSize
     * @return
     */
    public abstract MyWebSiteResult search(TbTrends tbTrends, int pageNum, int pageSize);

    /**
     * 添加动态
     * @param tbTrends
     * @return
     */
    public abstract MyWebSiteResult add(TbTrends tbTrends);

    /**
     * 修改动态
     * @param tbTrends
     * @return
     */
    public abstract MyWebSiteResult update(TbTrends tbTrends);

    /**
     * 删除动态
     * @param ids
     * @return
     */
    public abstract MyWebSiteResult deleteIds(Long[] ids);

    /**
     * 批量修改动态状态
     * @param ids
     * @param status
     * @return
     */
    public abstract MyWebSiteResult updateStatusIds(Long[] ids, String status);
}
